### PR TITLE
feat: db and collection name tags in traces

### PR DESCRIPTION
**Pull request by pboros**
DB and collection name in traces

### CODE REVIEW

**Comment by pboros** [comment]
This needed a refactoring on metrics. Changes related to that:
* Everything is gated on tracing.enabled, if it's enabled, there are traces and metrics, if not, they are not. We can make it more granular in the future if needed.
* Child traces always have a superset of it's parent trace's tags. 
* Metric tags and trace span tags are now from the same source and they are consistent.
* Tags always include the namespace, the db name and the collection name.

File: server/transaction/manager.go [hunk 1]
```diff
@@ -159,13 +158,15 @@ func newTxSessionWithMetrics(kv kv.KeyValueStore) (*TxSessionWithMetrics, error)
...
-	ctx, finishTracing = spanMeta.StartTracing(ctx, true)
-	defer finishTracing()
-	// TODO: better error handling (error in trace, etc)
-	ulog.E(f(ctx))
+	ctx = spanMeta.StartTracing(ctx, true)
+	err := f(ctx)
+	if err != nil {
+		spanMeta.FinishWithError(ctx, err)
+	}
+	_ = spanMeta.FinishTracing(ctx)
```
Comment (efirs) [inline server/transaction/manager.go:169]: is it ok to call this after FinishWithError?

File: server/transaction/manager.go [hunk 2]
```diff
@@ -159,13 +158,15 @@ func newTxSessionWithMetrics(kv kv.KeyValueStore) (*TxSessionWithMetrics, error)
...
+	// No counters here, just tracing, will be removed in the future
 	tags := metrics.GetFdbTags(ctx, name)
 	spanMeta := metrics.NewSpanMeta(metrics.TxManagerTracingServiceName, name, "tx_manager", tags)
-	ctx, finishTracing = spanMeta.StartTracing(ctx, true)
-	defer finishTracing()
-	// TODO: better error handling (error in trace, etc)
-	ulog.E(f(ctx))
+	ctx = spanMeta.StartTracing(ctx, true)
+	err := f(ctx)
+	if err != nil {
```
Comment (efirs) [inline server/transaction/manager.go:166]: `if err := f(ctx); err != nil {`

File: server/metrics/metrics.go [hunk 3]
```diff
@@ -41,11 +41,16 @@ var (
 )
 
 func GetGlobalTags() map[string]string {
-	return map[string]string{
+	res := map[string]string{
 		"service": util.Service,
 		"env":     config.GetEnvironment(),
-		"version": util.Version,
 	}
+	if util.Version == "" {
```
[the remark this answers is not in the document]
Comment (pboros) [reply server/metrics/metrics.go:48]: Great, I changed it like this.

File: server/metrics/tracing.go [hunk 4]
```diff
@@ -52,47 +91,79 @@ func (s *SpanMeta) GetSpanOptions() []tracer.StartSpanOption {
 	}
 }
 
-func (s *SpanMeta) StartTracing(ctx context.Context, childOnly bool) (context.Context, func()) {
+func (s *SpanMeta) AddTags(tags map[string]string) {
+	for k, v := range tags {
+		if _, exists := s.tags[k]; !exists {
```
[the remark this answers is not in the document]
Comment (pboros) [reply server/metrics/tracing.go:96]: Changed this as well.

File: server/transaction/manager.go (hunk 1 above)
Comment (pboros) [reply server/transaction/manager.go:169]: Good catch, the function should return after this.

File: server/transaction/manager.go (hunk 2 above)
Comment (pboros) [reply server/transaction/manager.go:166]: Changed it like this.

**Review by efirs — APPROVED** [review]

**Review by efirs — APPROVED** [review]

**Pull request merged**